### PR TITLE
Support .pac file on windows

### LINKED ISSUE #2: Support proxy.pac
Hi,

I would also like this library to support `proxy.pac` files (at least for windows).

To do this, I thought of using the [`WinHttpGetProxyForUrl` function](https://msdn.microsoft.com/en-us/library/windows/desktop/aa384097(v=vs.85).aspx) of the `Winhttp.dll` library.

However I don't have any experience with dll calling and I can't get it to work...
Do you have any idea?

```go
func main() {
	var winHttp = syscall.NewLazyDLL("Winhttp.dll")
	var open = winHttp.NewProc("WinHttpOpen")
	var close = winHttp.NewProc("WinHttpCloseHandle")
	var getProxyForUrl = winHttp.NewProc("WinHttpGetProxyForUrl")

	handle, _, err := open.Call(0, 0, 0, 0, 0)
	defer close.Call(handle)

	fmt.Println(err)

	ret, _, err := getProxyForUrl.Call(
		handle,
		uintptr(unsafe.Pointer(syscall.StringToUTF16Ptr("https://www.google.com"))),
		0,
		0,
	)

	fmt.Println(ret) // prints 0
	fmt.Println(err) // prints The parameter is incorrect.

}
```

### PR DESCRIPTION
**Pull request by oliverpool**
*Fix #2*

I finally got it to work :tada: 

You just need to do `proxy.FindProxyForURL(url)` on an `AutomaticProxyConf` object.

It's probably quite slow (no caching), but it works!

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+	go func() {
+		err := srv.Serve(listener.(*net.TCPListener))
+		if err != nil {
+			log.Println("HTTP Server Error - ", err)
+		}
+	}()
+
+	return listener, nil
+}
```
Comment (mattn) [inline utils.go]: please rename this to utils_test.go if those function is used for only tests.

File: pac_windows.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package ieproxy
+
+import (
+	"strings"
+	"syscall"
+	"unsafe"
+)
+
+type WINHTTP_AUTOPROXY_OPTIONS struct {
```
Comment (mattn) [inline pac_windows.go]: please rename this `tWINHTTP_AUTOPROXY_OPTIONS`, or move this to internal package with followings.

Comment (oliverpool) [inline pac_windows.go]: Only this one or `WINHTTP_PROXY_INFO` as well?
Should the `WINHTTP_AUTOPROXY_*` constants also have a lowercase letter prefix?

Comment (mattn) [inline pac_windows.go]: >Should the WINHTTP_AUTOPROXY_* constants also have a lowercase letter prefix?

Yes. If you don't want to change the name, please move those into internal/syscall/pac_windows.go

Comment (oliverpool) [inline pac_windows.go]: Done

**Comment by mattn** [comment]
Thank you!

**Pull request merged**